src/jet_client.rs

```diff
@@ -168,7 +161,8 @@ impl Future for JetMsgReader {
                 jet_proto::JET_MSG_HEADER_SIZE
             );
 
-            Ok(Async::NotReady)
+            cx.waker().clone().wake();
```

Comment:
Well, Indeed, it looks like it was a bug because normally we should call `wake` to ensure our future will be called if we are returning Pending manually (not while propagating it from another async function, but when actually returning if the first time in the call chain). In the tokio 0.1 it could be done with `futures::task::current().notify();` (e.g. in the old version of src/utils/association.rs we were doing so when call to try_lock of the non-async mutex returned `Err`. 
As I understand tokio 0.1 still _could_ poll the future when it's not notified anything, but it is not guaranteed, so maybe that's because we aren't faced any consequences of this bug yet.

Reply:
I understand, good catch!